src/fs/file.rs

```diff
@@ -303,19 +303,20 @@ impl File {
     ///
     /// # Return
     ///
-    /// The method returns the operation result and the same buffer value passed
-    /// as an argument.
+    /// The method returns the operation result and the buffer value passed
+    /// as an argument. The buffer is uniformly returned as a [`Slice`] view
+    /// covering the same range of the buffer as was passed to the method.
     ///
     /// If the method returns [`Ok(())`], then the read was successful.
     ///
     /// # Errors
     ///
     /// If this function encounters an "end of file" before completely filling
     /// the buffer, it returns an error of the kind [`ErrorKind::UnexpectedEof`].
-    /// The buffer is returned on error.
+    /// The buffer is returned on error as described above.
```

Comment:
Maybe add, "but the data that could be read will have been written to the buffer.

So it's clear that when there's an error, the buffer may have been written to somewhat. These write_all type functions are just helpers and can't be expected to be magic. If the caller cares, they should make their own copy beforehand.